micro-rdk/src/common/data_manager.rs

```diff
@@ -0,0 +1,402 @@
+use std::time::Duration;
+
+use crate::common::data_collector::{DataCollectionError, DataCollector};
+use crate::common::data_store::DataStore;
+use crate::proto::app::data_sync::v1::SensorData;
+
+use super::data_collector::ResourceMethodKey;
+use super::data_store::{DataStoreError, WriteMode};
+use async_io::Timer;
+use bytes::BytesMut;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum DataManagerError {
+    #[error("no data collectors in manager")]
+    NoCollectors,
+    #[error(transparent)]
+    CollectionError(#[from] DataCollectionError),
+    #[error(transparent)]
+    StoreError(#[from] DataStoreError),
+}
+
+pub struct DataManager<StoreType> {
+    collectors: Vec<DataCollector>,
+    store: StoreType,
+    sync_interval_ms: u64,
```

Comment:
Do we potentially gain anything by representing this as a `Duration`? Ditto for the various other places we use `_ms` / `u64`?